frontend/model/notifications/nativeNotification.js

```diff
@@ -4,50 +4,115 @@ import sbp from '@sbp/sbp'
 // NOTE: since these functions don't modify contract state, it should
 //       be safe to modify them without worrying about version conflicts.
 
-export async function requestNotificationPermission (force: boolean = false): Promise<null | string> {
-  if (typeof Notification === 'undefined') {
-    return null
-  }
+const handler = (statuses: string[]) => {
+  // For some reason, Safari seems to always return `'prompt'` with
+  // `Notification.permission` being correct.
+  const granted = statuses.every(status => status === 'granted') || (statuses.every(status => status === 'prompt') && Notification.permission === 'granted')
+  sbp('state/vuex/commit', 'setNotificationEnabled', granted)
```

Comment:
Claude has an interesting comment here. I'm not sure if it makes sense or not.

> The check of Notification.permission could be stale by the time it's read, since permission can be changed in another tab. A safer approach would be:
> 
> ```js
> const handler = async (statuses: string[]) => {
>   const currentPermission = await Notification.requestPermission()
>   const granted = statuses.every(status => status === 'granted') || 
>     (statuses.every(status => status === 'prompt') && currentPermission === 'granted')
>   sbp('state/vuex/commit', 'setNotificationEnabled', granted)
> }
> ```

Reply:
This is supposed to the handler to detect permission changes. Making it async or making it check again isn't how it's supposed to work.